setup.py

```diff
@@ -258,25 +282,32 @@ def add_dir_name(dir_name, filenames):
         for opts in [swig_opts, extra_compile_args]:
             opts.extend(["-I%s/" % mod.src, "-I%s/" % mod.swig])
 
-        # Add compile-time location of the dependency library
-        library_dirs.append(os.path.join(mod.build_dir, mod.build))
-
         # Because setuptools produces shared object files with non-standard
-        # names (i.e. not lib<name>.so) we prepend with colon to specify the
-        # name directly
-        libraries.append(":" + mod.lib_filename)
-
-        # Make sure that the runtime linker can find shared object dependencies
-        # by using the relative path to the dependency library. $ORIGIN refers
-        # to the location of the current shared object file at runtime
-        runtime_library_dirs.append(
-            "\$ORIGIN/%s" % os.path.relpath(mod.build, swig_path.build))
-
+        # names (i.e. not lib<name>.so) we specify the full library path
+        extra_link_args.append(os.path.abspath(
+            os.path.join(build_dir, mod.build, mod.lib_filename)))
+
+        # Make sure that the runtime linker can find shared object
+        # dependencies by using the relative path to the dependency library.
+        if platform.system() == 'Linux':
+            # $ORIGIN refers to the location of the current shared object file
+            # at runtime
+            runtime_library_dirs.append(
+                "\$ORIGIN/%s" % os.path.relpath(mod.build, swig_path.build))
+        else: # Assuming non-Windows builds for now
+            # For OSX builds we use @loader_path instead
+            extra_link_args.append(
+                "-Wl,-rpath,%s" % '@loader_path/%s' % os.path.relpath(mod.build, swig_path.build)
```

Comment:
`os.path.relpath(mod.build, swig_path.build)` can be given a name (`rel_path_to_shared_object` ?) and shared by osx and linus